src/CurrentRoute/View/Helper/CurrentRoute.php

```diff
@@ -30,10 +30,33 @@ public function __construct($controller, $action, $route, $module)
         }
 
         $this->module     = $module[0];
-        $this->controller = array_pop($controller);
-        $this->action     = $action;
+        $this->controller = $this->removeControllerActionSuffix(array_pop($controller));
+        $this->action     = $this->removeControllerActionSuffix($action);
         $this->route      = $route;
     }
+    
+    /**
+     * @param $name
+     * @return string $name
+     * remove xxxAction and xxxController suffix from names
+     */
+    private function removeControllerActionSuffix($name)
+    {
+        
+        if (strpos($name,'controller') !== false )
+        {
+            $name = substr($name,0, strpos($name,'controller') );
+        }
+
+        if (strpos($name,'action') !== false )
```

Comment:
as I can checked there is no need to filter 'Action' since is not passed in constructor with the suffix.